src/tests/suite/daos_checksum.c

```diff
@@ -261,10 +269,60 @@ io_with_server_side_verify(void **state)
 			     &ctx.update_iod, &ctx.update_sgl, NULL);
 	assert_int_equal(rc, -DER_CSUM);
 	cleanup_cont_obj(&ctx);
+	unset_csum_fi();
+
+	/**5. Data corruption. Update should fail due CRC mismatch */
+	set_client_data_corrupt_fi();
+	setup_cont_obj(&ctx, DAOS_PROP_CO_CSUM_CRC64, false, 0, OC_SX);
+	rc = daos_obj_update(ctx.oh, DAOS_TX_NONE, 0, &ctx.dkey, 1,
+			     &ctx.update_iod, &ctx.update_sgl, NULL);
+	assert_int_equal(rc, 0);
+	cleanup_cont_obj(&ctx);
 
+	setup_cont_obj(&ctx, DAOS_PROP_CO_CSUM_CRC64, true, 0, OC_SX);
+	rc = daos_obj_update(ctx.oh, DAOS_TX_NONE, 0, &ctx.dkey, 1,
+			     &ctx.update_iod, &ctx.update_sgl, NULL);
+	assert_int_equal(rc, -DER_CSUM);
+	cleanup_cont_obj(&ctx);
 	unset_csum_fi();
+	cleanup_data(&ctx);
+}
+
+static void
+test_server_data_corruption(void **state)
+{
+	test_arg_t	*arg = *state;
+	struct csum_test_ctx	 ctx = {0};
+	int			 rc;
+
+	setup_from_test_args(&ctx, *state);
+	setup_cont_obj(&ctx, DAOS_PROP_CO_CSUM_CRC64, false, 1024*8, OC_SX);
+
+	/**1. Simple server data corruption after RDMA */
+	setup_multiple_extent_data(&ctx);
+	/** Set the Server data corruption flag */
+	rc = daos_mgmt_set_params(arg->group, -1, DMG_KEY_FAIL_LOC,
+				DAOS_CHECKSUM_SDATA_CORRUPT | DAOS_FAIL_ALWAYS,
+				0, NULL);
+	assert_int_equal(rc, 0);
+	/** Adding delay for functional testing under VM */
```

Comment:
why is testing needed on VM? 2 minutes seems like a long time

Reply:
Initially, I tried 60 seconds and tests was failing at intermittent intervals due to the server corrupt flag not set... Also, I noticed rebuild/aggregation happening while the test is running on the server nodes. It is possible fault injection flag is ignored or skipped during the rebuild/aggregation process. I am not 100 percent sure. But it looks like that maybe the issue. Adding a 120 second delay made test reliable on the CI since all rebuild/aggregation completes within timeframe.

Reply:
Removed.